packages/@blockprotocol/graph/src/types/subgraph/edges/outward-edge-alias.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * A collection of 'aliases' which describe various variants of outward edges in more accessible-forms
+ */
+
+import { EntityIdAndTimestamp, OutwardEdge } from "./outward-edge";
+
+/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
```

Comment:
Can you give me an example of one where it doesn't?

Reply:
What I mean is you can change it to be whatever you want, you can make the string "foo" if you want. Which is unfortunate if we were to refactor `KnowledgeGraphOutwardEdge`, as this wouldn't be picked up, and wouldn't cause type errors (except in usage).

Again I want to be able to say `OutgoingLinkEdge satisfies KnowledgeGraphOutwardEdge`. It should be possible to actually define a subtype.

Reply:
Got it – this is maybe as close as you can get
```ts
type SubType<Type extends Record<string, unknown>, Sub extends Type> = {
  [Key in keyof Type]: Sub[Key];
};

export type OutgoingLinkEdge = SubType<
  KnowledgeGraphOutwardEdge,
  {
    reversed: true;
    kind: "asdad"; // error, 
    rightEndpoint: EntityIdAndTimestamp;
  }
>;
```
This will not prevent additional properties being assigned, which is hard to eliminate everywhere in TS (see [here](https://stackoverflow.com/a/62099607/17217717)).

Reply:
Hm, I wish the IDE understood this a little better :( 

You don't get any autocomplete for example.

<img width="388" alt="image" src="https://user-images.githubusercontent.com/25749103/207290542-e21aa8d0-d950-41a6-97ea-a806f8c8321f.png">

And refactoring (e.g. changing `reversed` to `test`):
<img width="744" alt="image" src="https://user-images.githubusercontent.com/25749103/207291132-3d4dfbdf-5ab6-4e5c-8fef-7413ff9a2ced.png">

Doesn't get picked up:
<img width="367" alt="image" src="https://user-images.githubusercontent.com/25749103/207291225-04b58fd3-f36b-4969-abbe-ec19ce8a96e5.png">

At least it does throw an error so I guess it's better than before. I'll add it, I just wish there was native support :(

Thanks again

Reply:
[Add Subtype util and constrain link aliases](https://github.com/blockprotocol/blockprotocol/pull/820/commits/e3e6394788fa9574b8ae5eede0c5a8311bd6d2d5)